internal/gatewayapi/helpers.go

```diff
@@ -194,11 +195,19 @@ func ValidateHTTPRouteFilter(filter *v1beta1.HTTPRouteFilter) error {
 		switch {
 		case filter.ExtensionRef == nil:
 			return errors.New("extensionRef field must be specified for an extended filter")
-		case string(filter.ExtensionRef.Group) != egv1a1.GroupVersion.Group:
-			return fmt.Errorf("invalid group; must be %s", egv1a1.GroupVersion.Group)
-		case string(filter.ExtensionRef.Kind) != egv1a1.AuthenticationFilterKind:
-			return fmt.Errorf("invalid kind; must be %s", egv1a1.AuthenticationFilterKind)
+		// extensionRef references Envoy Gateway API group
+		case string(filter.ExtensionRef.Group) == egv1a1.GroupVersion.Group:
+			switch {
+			case string(filter.ExtensionRef.Kind) != egv1a1.AuthenticationFilterKind:
```

Comment:
preference: usage of an inner switch here when it is doing if/else logic is confusing.